### ISSUE TITLE
OneSignalDbHelper.java line 107 | com.onesignal.al.b | Fatal Exception: android.database.sqlite.SQLiteException: not an error (Sqlite code 0), (OS error - 2:No such file or directory)

### ISSUE THREAD

**Issue by bytekode** [OneSignal/OneSignal-Android-SDK#594]
**Description:**

I am using OneSignal Android SDK and this crash is occurring very frequently and here is link http://crashes.to/s/101f00453ee

**Environment**
1. Android SDK version **3.9.2**
2. build.gradle entry  **implementation 'com.onesignal:OneSignal:3.9.2'**

**Steps to Reproduce Issue:**

Unfortunately I couldn't reproduce this issue and is occurring in production environment.

**Anything else:**
```java
Fatal Exception: android.database.sqlite.SQLiteException: not an error (Sqlite code 0), (OS error - 2:No such file or directory)
       at android.database.sqlite.SQLiteConnection.nativeRegisterLocalizedCollators(SQLiteConnection.java)
       at android.database.sqlite.SQLiteConnection.setLocaleFromConfiguration(SQLiteConnection.java:382)
       at android.database.sqlite.SQLiteConnection.open(SQLiteConnection.java:228)
       at android.database.sqlite.SQLiteConnection.open(SQLiteConnection.java:203)
       at android.database.sqlite.SQLiteConnectionPool.openConnectionLocked(SQLiteConnectionPool.java:511)
       at android.database.sqlite.SQLiteConnectionPool.open(SQLiteConnectionPool.java:194)
       at android.database.sqlite.SQLiteConnectionPool.open(SQLiteConnectionPool.java:183)
       at android.database.sqlite.SQLiteDatabase.openInner(SQLiteDatabase.java:880)
       at android.database.sqlite.SQLiteDatabase.open(SQLiteDatabase.java:864)
       at android.database.sqlite.SQLiteDatabase.openDatabase(SQLiteDatabase.java:767)
       at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:250)
       at android.database.sqlite.SQLiteOpenHelper.getReadableDatabase(SQLiteOpenHelper.java:199)
       at com.onesignal.al.b(OneSignalDbHelper.java:107)
       at com.onesignal.l.a(GenerateNotification.java:480)
       at com.onesignal.l.d(GenerateNotification.java:354)
       at com.onesignal.l.a(GenerateNotification.java:106)
       at com.onesignal.q.a(NotificationBundleProcessor.java:96)
       at com.onesignal.q.a(NotificationBundleProcessor.java:75)
       at com.onesignal.GcmIntentService.onHandleIntent(GcmIntentService.java:60)
       at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:67)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:156)
       at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by jkasten2** [comment]
@bytekode It looks like the crash is related to OneSignal not able to create a database for a list of notifications your app has received. It looks like this crash started happening for you mostly on versionCode 101 of your app. Could you summarize what changed in this version?

**Comment by bytekode** [comment]
// In Android Application class onCreate method

// Code snippet since version code < 101
OneSignal.startInit(this)
                .inFocusDisplaying(OneSignal.OSInFocusDisplayOption.Notification)
                .setNotificationOpenedHandler(new OpenHandler(getApplicationContext()))
                .unsubscribeWhenNotificationsAreDisabled(true)
                .init();

// Added in version code 101
OneSignal.idsAvailable(new OneSignal.IdsAvailableHandler() {
            @Override
            public void idsAvailable(String userId, String registrationId) {
                Log.e(TAG, "idsAvailable: userId" + userId);
                if (registrationId != null) {
                    // Store the registration id in shared preferences for future use
                    Helper.setRegistrationID(getApplicationContext(), registrationId);
                    Log.e(TAG, "registrationId:" + registrationId);
                } else {
                    Log.e(TAG, "registrationId NOT AVAILABLE");
                }
            }
        });

**Comment by jkasten2** [comment]
@bytekode Thanks for the code changes from your 100 to 101 version. `OneSignal.idsAvailable` will read from shared preferences and not SQL so this shouldn't be related to this crash.

The crash stack is happening when trying to generate an updated summary notification after receiving one. Are you sending different notifications if uses have updated to version 101 of your app?

**Comment by jkasten2** [comment]
Closing due to inactivity. Please provide more details and we can try to look into this further. Thanks.
